syntax = "proto2";

package classification.protos;


message RestoredVariableScope {
    optional string name_in_checkpoint = 1[default=""];
    required string name_in_model      = 2;
}

// Message for configuring DetectionModel training jobs (train.py).
message TrainConfig {


  // Data augmentation options.
//  repeated PreprocessingStep data_augmentation_options = 2;

  // How frequently to keep checkpoints.
  optional uint32 keep_checkpoint_every_n_minutes = 4 [default=1];

  // Optimizer used to train the DetectionModel.
  //optional Optimizer optimizer = 5;

  // If greater than 0, clips gradients by this value.
  //optional float gradient_clipping_by_norm = 6 [default=0.0];

  //initial checkpoint to fine_tune from
  optional string fine_tune_checkpoint = 7 [default=""];

  //use has the option of either specifying which to exclude from the fine-tune
  //or which to include
  repeated string exclude_from_fine_tune = 8;

  repeated RestoredVariableScope include_from_fine_tune =9;

  //where to store/save training checkpoint (used for restoring too)
  optional string from_classification_checkpoint = 10 [default=""];

  // Number of steps to train the Model for. If 0, will train the model
  // indefinitely.
  optional uint32 num_steps = 11 [default=0];

  // Variables or Scopes that should not be updated during training.
  repeated string scopes_or_variables_to_train = 12;

  //update ops to use (specify "all" to use all update_ops)
  repeated string scopes_or_names_for_update_ops = 13;

  //whether to evaluate while training
  optional bool eval_while_training = 14 [default=false];

  //number of steps to log at
  optional int32 log_frequency = 15[default=10];

  optional float learning_rate = 16[default=0.001];
}
